### PR TITLE
User Routes

### PR DESCRIPTION
**Pull request by vincetiu8**

### CODE REVIEW

File: src/controllers/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+  const user = req.user as User;
+
+  if (!user) {
+    return res.status(401).json({ error: 'Not logged in' });
+  }
+
+  const profileUpdate = await updateUser({
+      user_id: user.user_id,
+      password,
```
Comment (joshchen984) [inline src/controllers/user.controller.ts:150]: Can you hash password before saving to database

File: src/config/configPassport.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      {
+        usernameField: 'email', // Treat email field in request as username
+      },
+      verifyLocalUser,
+    ),
+  );
+
+  // Set up serialization and deserialization of user objects
+  passport.serializeUser((user: any, done: any) => {
+    done(null, user._id);
```
Comment (joshchen984) [inline src/config/configPassport.ts]: Does the user have a ._id field cause we're storing it as user_id? Should we store user_id instead?

File: src/controllers/user.controller.ts (hunk 1 above)
Comment (joshchen984) [reply src/controllers/user.controller.ts:150]: Nvm password already hashed

File: src/config/configPassport.ts (hunk 2 above)
Comment (vincetiu8) [reply src/config/configPassport.ts]: You're right, fixed.

**Pull request merged**